code/game/mecha/mecha_actions.dm

```diff
@@ -0,0 +1,281 @@
+	button_icon_state = "mech_internals_[chassis.use_internal_tank ? "on" : "off"]"
+	chassis.occupant_message("Now taking air from [chassis.use_internal_tank?"internal airtank":"environment"].")
+	chassis.log_message("Now taking air from [chassis.use_internal_tank?"internal airtank":"environment"].")
+	UpdateButtonIcon()
+
+/datum/action/innate/mecha/mech_cycle_equip
+	name = "Cycle Equipment"
+	button_icon_state = "mech_cycle_equip_off"
+
+/datum/action/innate/mecha/mech_cycle_equip/Activate()
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+
+	var/list/available_equipment = list()
+	for(var/obj/item/mecha_parts/mecha_equipment/M in chassis.equipment)
+		if(M.selectable)
+			available_equipment += M
+
+	if(available_equipment.len == 0)
+		chassis.occupant_message("No equipment available.")
+		return
+	if(!chassis.selected)
+		chassis.selected = available_equipment[1]
+		chassis.occupant_message("You select [chassis.selected]")
+		send_byjax(chassis.occupant,"exosuit.browser","eq_list",chassis.get_equipment_list())
+		button_icon_state = "mech_cycle_equip_on"
+		UpdateButtonIcon()
+		return
+	var/number = 0
+	for(var/A in available_equipment)
+		number++
+		if(A == chassis.selected)
+			if(available_equipment.len == number)
+				chassis.selected = null
+				chassis.occupant_message("You switch to no equipment")
+				button_icon_state = "mech_cycle_equip_off"
+			else
+				chassis.selected = available_equipment[number+1]
+				chassis.occupant_message("You switch to [chassis.selected]")
+				button_icon_state = "mech_cycle_equip_on"
+			send_byjax(chassis.occupant,"exosuit.browser","eq_list",chassis.get_equipment_list())
+			UpdateButtonIcon()
+			return
+
+
+/datum/action/innate/mecha/mech_toggle_lights
+	name = "Toggle Lights"
+	button_icon_state = "mech_lights_off"
+
+/datum/action/innate/mecha/mech_toggle_lights/Activate()
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+	chassis.lights = !chassis.lights
+	if(chassis.lights)
+		chassis.AddLuminosity(chassis.lights_power)
+		button_icon_state = "mech_lights_on"
+	else
+		chassis.AddLuminosity(-chassis.lights_power)
+		button_icon_state = "mech_lights_off"
+	chassis.occupant_message("Toggled lights [chassis.lights?"on":"off"].")
+	chassis.log_message("Toggled lights [chassis.lights?"on":"off"].")
+	UpdateButtonIcon()
+
+/datum/action/innate/mecha/mech_view_stats
+	name = "View Stats"
+	button_icon_state = "mech_view_stats"
+
+/datum/action/innate/mecha/mech_view_stats/Activate()
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+	chassis.occupant << browse(chassis.get_stats_html(), "window=exosuit")
+
+
+/datum/action/innate/mecha/strafe
+	name = "Toggle Strafing"
+	button_icon_state = "strafe"
+
+/datum/action/innate/mecha/strafe/Activate()
+	if(!owner || !chassis || chassis.occupant != owner)
+		return
+
+	chassis.toggle_strafe()
+
+/obj/mecha/AltClick(mob/living/user)
+	if(user == occupant)
+		toggle_strafe()
+
+/obj/mecha/proc/toggle_strafe()
```

Comment:
Some comment in this PR telling us what you did and didn't change in this file would've been nice.